.github/workflows/perfcheck.yml

```diff
@@ -0,0 +1,205 @@
+        kill %1
+        ./vegeta report -type=text $REPORT
+        ./vegeta report -type=json $REPORT | jq --arg name "$NAME" -r -f scripts/ci/vegeta_report_to_csv.jq >> graph.csv
+
+    - name: Run Apache benchmark
+      run: |
+        NAME=$(apache2 -v | sed 's/.*: //' | sed 's/ .*//' | sed 's!/! !')
+        REPORT=report_apache.bin
+        apache2 -D FOREGROUND -f `pwd`/.github/workflows/config/benchmarking/apache.conf &
+        sleep 2
+        echo "GET ${{ matrix.url }}" | ./vegeta attack -name "$NAME" ${{ env.VEGETA_FLAGS }} > $REPORT
+        kill %1
+        ./vegeta report -type=text $REPORT
+        ./vegeta report -type=json $REPORT | jq --arg name "$NAME" -r -f scripts/ci/vegeta_report_to_csv.jq >> graph.csv
+
+    - name: Run lighttpd benchmark
+      run: |
+        NAME=$(lighttpd -v | sed 's/ .*//' | sed 's!/! !')
+        REPORT=report_lighttpd.bin
+        lighttpd -D -f `pwd`/.github/workflows/config/benchmarking/lighttpd.conf &
+        sleep 2
+        URL="${{ matrix.url }}"
+        if [[ $URL == *"_precompressed"* ]]; then
+          # lighttpd does not really support pre-compressed files, request them explicitly
+          URL="${{ matrix.url }}.gz"
+        fi
+        echo "GET $URL" | ./vegeta attack -name "$NAME" ${{ env.VEGETA_FLAGS }} > $REPORT
+        kill %1
+        ./vegeta report -type=text $REPORT
+        ./vegeta report -type=json $REPORT | jq --arg name "$NAME" -r -f scripts/ci/vegeta_report_to_csv.jq >> graph.csv
+
+    - name: Run nginx benchmark
+      run: |
+        NAME=$(nginx -v 2>&1 | sed 's/.*: //' | sed 's/ .*//' | sed 's!/! !')
+        REPORT=report_nginx.bin
+        nginx -c `pwd`/.github/workflows/config/benchmarking/nginx.conf &
+        sleep 2
+        echo "GET ${{ matrix.url }}" | ./vegeta attack -name "$NAME" ${{ env.VEGETA_FLAGS }} > $REPORT
+        kill %1
+        ./vegeta report -type=text $REPORT
+        ./vegeta report -type=json $REPORT | jq --arg name "$NAME" -r -f scripts/ci/vegeta_report_to_csv.jq >> graph.csv
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+
+    - name: Install pipx
+      run: pip install pipx
+
+    - name: Produce plot
+      run: |
+        ./vegeta plot --title "${{ matrix.title }}" report_*.bin > detailed_plot.html
+        cat graph.csv
+        cat graph.csv | sed -e 's/,/|/g' -e 's/"//g' -e 's/^/|/' -e 's/$/|/' -e '2 i |-|-|-|-|-|-|-|-|' >> $GITHUB_STEP_SUMMARY
+        pipx run graph-cli --barh --bar-label --xlabel '' --ylabel 'Latency (ms)' --title "${{ matrix.title }}" --figsize 1200x700 --fontsize 10 --ytick-angle 45 --ytick-align top -o overview.png graph.csv
+
+    - name: Upload benchmark plots
+      uses: actions/upload-artifact@v4
+      with:
+        name: Benchmark plots (${{ matrix.title }})
+        path: |
+          detailed_plot.html
+          overview.png
```

Comment:
For the artifacts produced by the CI matrix, Is there a possibility to collect them and compress them to a single file rather than display the 20 individually? I think if someone is interested in looking at the details, then can always download the zip file. What do you think?

Reply:
I think that the artifact upload added a merge feature recently, this should work here. But the ZIP file will be huge for the binary reports. I’d still keep the plots and the reports as two separate artifacts, so that the plots stay at least somewhat downloadable.

Reply:
Merging took 2 minutes 30 seconds for the binary reports (500 MB). So I removed uploading them altogether, all the other data available should do in mostly scenarios.